packages/block-library/src/table-of-contents/index.php

```diff
@@ -0,0 +1,174 @@
+<?php
+/**
+ * Server-side rendering of the `core/table-of-contents` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Registers the `core/table-of-contents` block on the server.
+ *
+ * @since 6.4.0
+ */
+function register_block_core_table_of_contents() {
+	// @todo: Register meta for all post types that use Block Editor.
+	foreach ( array( 'post', 'page' ) as $post_type ) {
+		register_post_meta(
+			$post_type,
+			'table_of_contents',
```

Comment:
I thought the same but noticed that the Footnotes block wasn't using the prefix. I'll add a `wp_` prefix, just in case.

Reply:
From my experience, the `wp_` prefix is very common too for custom meta in custom plugins (for some weird reason). I'd recommend `_core_` or `core_`, WDYT?

Reply:
Let's go with `core_`. Leading underscore in meta prefix indicates it's private status and core might perform extra checks. It's not needed here as meta is public.

Reply:
`core_` sounds good to me, but...

Just to clarify, the reason I suggested prefixing it with an underscore is that we actually might want to keep it private. Public meta will appear in the Custom Fields section below the editor, and that, to me, is undesired:

![Screenshot 2023-09-12 at 15 42 36](https://github.com/WordPress/gutenberg/assets/8436925/4b45e822-68b3-4c46-9662-1ec8ebec1e0a)

It appears there are at least a few downsides to that:
- it might bring confusion to the user
- the user might edit/delete it by accident
- it doesn't update in realtime - if you add a heading, it doesn't immediately recompute)

Reply:
I decided to go with a public meta for the moment. As I was worried, protecting a meta key enabled additional checks, like capability check, before updating the meta. I don't think it is needed here.

Hiding meta values from Custom Fields doesn't prevent users from deleting the meta; it's a "security through obscurity" IMO.

Reply:
> As I was worried, protecting a meta key enabled additional checks, like capability check, before updating the meta.

Could you please link to where that capability check is, just to make sure we're on the same page about it?

Reply:
There are a few places like [register_meta](https://developer.wordpress.org/reference/functions/register_meta/) and [map_meta_cap](https://developer.wordpress.org/reference/functions/map_meta_cap/). Search for is_protected_meta().

You can also protect the meta key and try saving the changes to a heading. You'll get a permission error.

Reply:
Makes sense, thanks for elaborating. 

I guess it's easy enough to add `'auth_callback' => '__return_true'` to the `$args` of `register_post_meta()` and you'll bypass that capability check easily. If you want to be more precise, you could also return a function that returns `current_user_can( 'edit_post', $post_id )`.

If you ask me, the bigger win is not showing the serialized data to the end user who might see it and change it inadvertently.

However, I'm happy to leave the last decision to you.

Given that this works with a visible meta for the footnotes block as well, I think we can leave it as-is, and revisit it at a later time (although if we switch to private meta later, this will unfortunately clutter the database with old public unnecessary meta).